pinot-common/src/main/java/org/apache/pinot/common/utils/TlsUtils.java

```diff
@@ -255,6 +257,14 @@ public static URL makeKeyStoreUrl(String storePath)
     return inputUri.toURL();
   }
 
+  public static SSLContext getSslContext() {
+    return _sslContext != null ? _sslContext : SSLContexts.createDefault();
```

Comment:
i dont know whether this needs to be synchronized as it is only generated once.

the only way to modify it is via setSslContext, which should also be statically accessed when the secure environment constructs at startup time).

Reply:
Also CC @richardstartin to give suggestions on how to set static variables

Reply:
yeah I was thinking about using lazy static initialization similar to https://stackoverflow.com/a/7420714 since it is static thread safe. 

However the current way how TLSUtils install the TLSSpec is non-static. so it might not be possible to acquire the TLSSpecs at jvm start up time.

Reply:
I suggest putting it in a holder class:

```java
private static final AtomicReference<SSLContext> OVERRIDE = new AtomicReference<>();

public static void registerOverride(SSLContext override) {
    if (!OVERRIDE.compareAndSet(null, override)) {
       // warn that something else beat the caller here
    }
}

private static final class Holder {
    static final SSLContext SSL_CONTEXT = OVERRIDE.get() == null ? SSLContexts.createDefault() : OVERRIDE.get();
}
```

And then reference it via `Holder.SSL_CONTEXT` everywhere. This has a few advantages:

* registration is thread safe
* mutable until first use
* synchronization, at most once initialisation guaranteed by the classloader
* after initialisation, the `SSLContext` is constant which can drive optimisations like constant folding, which may or may not be important

Reply:
yeah. this is what I meant previously (less the `OVERRIDE` part which brilliantly solve the tls keystore init issue). thanks @richardstartin !